### ISSUE TITLE
Segfault when installing Chef on Windows 7

### ISSUE THREAD

**Issue by tknerr** [chef/chef#3609]
Thought it might be #3606, but even after locking FFI to 1.9.8 I'm still getting this segfault when I `bundle install` in this [Gemfile](https://github.com/tknerr/sample-toplevel-cookbook/blob/master/Gemfile):

Note: the Gemfile above uses chef 12.3.0 and latest ffi, the output below was with `gem 'chef', '12.4.0'` and `gem 'ffi', '1.9.8'`, but doesn't make a difference...

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    W:/tools/chefdk/embedded/bin/ruby.exe -rubygems W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/bin/rake RUBYARCHDIR=W:/home/.chefdk/gem/ruby/2.1.0/extensions/x86-mingw32/2.1.0/chef-12.4.0-universal-mingw32 RUBYLIBDIR=W:/home/.chefdk/gem/ruby/2.1.0/extensions/x86-mingw32/2.1.0/chef-12.4.0-universal-mingw32
checking for windmc... yes
checking for windres... yes
checking for gcc... yes
windmc chef-log.man
windres -i chef-log.rc -o resource.o
gcc -o chef-log.dll -shared resource.o
W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/eventlog.rb:211: [BUG] Segmentation fault
ruby 2.1.5p273 (2014-11-13 revision 48405) [i386-mingw32]

-- Control frame information -----------------------------------------------
c:0028 p:---- s:0119 e:000118 CFUNC  :RegCreateKeyEx
c:0027 p:0206 s:0107 e:000106 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/eventlog.rb:211
c:0026 p:0063 s:0095 e:000094 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/chef-12.4.0-universal-mingw32/ext/win32-eventlog/Rakefile:45 [FINISH]
c:0025 p:---- s:0091 e:000090 CFUNC  :call
c:0024 p:0028 s:0086 e:000085 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240 [FINISH]
c:0023 p:---- s:0083 e:000082 CFUNC  :each
c:0022 p:0113 s:0080 e:000079 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235
c:0021 p:0075 s:0076 e:000075 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:179
c:0020 p:0014 s:0074 e:000073 METHOD W:/tools/chefdk/embedded/lib/ruby/2.1.0/monitor.rb:211
c:0019 p:0025 s:0071 e:000070 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172
c:0018 p:0024 s:0064 e:000063 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:201 [FINISH]
c:0017 p:---- s:0060 e:000059 CFUNC  :each
c:0016 p:0031 s:0057 e:000056 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:199
c:0015 p:0062 s:0052 e:000051 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:178
c:0014 p:0014 s:0050 e:000049 METHOD W:/tools/chefdk/embedded/lib/ruby/2.1.0/monitor.rb:211
c:0013 p:0025 s:0047 e:000046 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172
c:0012 p:0036 s:0040 e:000039 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:165
c:0011 p:0033 s:0035 e:000034 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:150
c:0010 p:0009 s:0028 e:000027 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106 [FINISH]
c:0009 p:---- s:0025 e:000024 CFUNC  :each
c:0008 p:0039 s:0022 e:000021 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106
c:0007 p:0025 s:0020 e:000019 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:115
c:0006 p:0007 s:0016 e:000015 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:100
c:0005 p:0019 s:0013 e:000012 BLOCK  W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:78
c:0004 p:0006 s:0011 e:000010 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:176
c:0003 p:0007 s:0007 e:000006 METHOD W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:75
c:0002 p:0040 s:0004 E:00207c EVAL   W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/bin/rake:33 [FINISH]
c:0001 p:0000 s:0002 E:00245c TOP    [FINISH]

W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/bin/rake:33:in `<main>'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
W:/tools/chefdk/embedded/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:199:in `invoke_prerequisites'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:199:in `each'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
W:/tools/chefdk/embedded/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
W:/home/.chefdk/gem/ruby/2.1.0/gems/chef-12.4.0-universal-mingw32/ext/win32-eventlog/Rakefile:45:in `block in <top (required)>'
W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/eventlog.rb:211:in `add_event_source'
W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/eventlog.rb:211:in `RegCreateKeyEx'

-- C level backtrace information -------------------------------------------
C:\Windows\SysWOW64\ntdll.dll(ZwWaitForSingleObject+0x15) [0x7730F891]
C:\Windows\syswow64\kernel32.dll(WaitForSingleObjectEx+0x43) [0x74DC1194]
C:\Windows\syswow64\kernel32.dll(WaitForSingleObject+0x12) [0x74DC1148]
W:\tools\chefdk\embedded\bin\msvcrt-ruby210.dll(rb_vm_bugreport+0xa7) [0x6D394A17]
W:\tools\chefdk\embedded\bin\msvcrt-ruby210.dll(rb_name_err_mesg_new+0x69f) [0x6D248F7F]
W:\tools\chefdk\embedded\bin\msvcrt-ruby210.dll(rb_bug+0x2e) [0x6D249D5E]
W:\tools\chefdk\embedded\bin\msvcrt-ruby210.dll(rb_check_safe_str+0x34b) [0x6D31486B]
 [0x00401866]
C:\Windows\SysWOW64\ntdll.dll(RtlKnownExceptionFilter+0xb7) [0x773636E7]

-- Other runtime information -----------------------------------------------

* Loaded script: W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/bin/rake

* Loaded features:

    0 enumerator.so
    1 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/enc/encdb.so
    2 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/enc/iso_8859_1.so
    3 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/enc/trans/transdb.so
    4 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/rbconfig.rb
    5 thread.rb
    6 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/thread.so
    7 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/compatibility.rb
    8 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/defaults.rb
    9 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/deprecate.rb
   10 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/errors.rb
   11 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/version.rb
   12 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/requirement.rb
   13 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/platform.rb
   14 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/basic_specification.rb
   15 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/stub_specification.rb
   16 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/util/stringio.rb
   17 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb
   18 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/exceptions.rb
   19 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/io/console.so
   20 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/user_interaction.rb
   21 W:/tools/chefdk/embedded/lib/ruby/2.1.0/i386-mingw32/etc.so
   22 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/config_file.rb
   23 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/defaults/operating_system.rb
   24 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_gem.rb
   25 W:/tools/chefdk/embedded/lib/ruby/2.1.0/monitor.rb
   26 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb
   27 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems.rb
   28 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/ubygems.rb
   29 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/dependency.rb
   30 W:/tools/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/path_support.rb
   31 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/version.rb
   32 W:/tools/chefdk/embedded/lib/ruby/2.1.0/fileutils.rb
   33 W:/tools/chefdk/embedded/lib/ruby/2.1.0/singleton.rb
   34 W:/tools/chefdk/embedded/lib/ruby/2.1.0/optparse.rb
   35 W:/tools/chefdk/embedded/lib/ruby/2.1.0/ostruct.rb
   36 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/ext/module.rb
   37 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/ext/core.rb
   38 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/ext/string.rb
   39 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/early_time.rb
   40 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/late_time.rb
   41 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/ext/time.rb
   42 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/alt_system.rb
   43 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/win32.rb
   44 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/linked_list.rb
   45 W:/tools/chefdk/embedded/lib/ruby/2.1.0/open3.rb
   46 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/cpu_counter.rb
   47 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/scope.rb
   48 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task_argument_error.rb
   49 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/rule_recursion_overflow_error.rb
   50 W:/tools/chefdk/embedded/lib/ruby/2.1.0/shellwords.rb
   51 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task_manager.rb
   52 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/cloneable.rb
   53 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/file_utils.rb
   54 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/file_utils_ext.rb
   55 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/pathmap.rb
   56 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/file_list.rb
   57 W:/tools/chefdk/embedded/lib/ruby/2.1.0/set.rb
   58 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/promise.rb
   59 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/thread_pool.rb
   60 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/private_reader.rb
   61 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/thread_history_display.rb
   62 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/trace_output.rb
   63 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/application.rb
   64 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/rake_module.rb
   65 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/pseudo_status.rb
   66 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task_arguments.rb
   67 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/invocation_chain.rb
   68 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/invocation_exception_mixin.rb
   69 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/task.rb
   70 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/file_task.rb
   71 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/file_creation_task.rb
   72 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/multi_task.rb
   73 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/dsl_definition.rb
   74 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/default_loader.rb
   75 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/name_space.rb
   76 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake/backtrace.rb
   77 W:/home/.chefdk/gem/ruby/2.1.0/gems/rake-10.4.2/lib/rake.rb
   78 W:/tools/chefdk/embedded/lib/ruby/2.1.0/mkmf.rb
   79 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/windows/constants.rb
   80 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/2.1/ffi_c.so
   81 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/platform.rb
   82 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/types.rb
   83 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/library.rb
   84 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/errno.rb
   85 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/pointer.rb
   86 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/memorypointer.rb
   87 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/struct_layout_builder.rb
   88 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/struct.rb
   89 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/union.rb
   90 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/managedstruct.rb
   91 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/callback.rb
   92 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/io.rb
   93 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/autopointer.rb
   94 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/variadic.rb
   95 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/enum.rb
   96 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/ffi.rb
   97 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi.rb
   98 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/windows/structs.rb
   99 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/windows/functions.rb
  100 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/eventlog.rb

[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: http://www.ruby-lang.org/bugreport.html


This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.

rake failed, exit code 3

Gem files will remain installed in W:/home/.chefdk/gem/ruby/2.1.0/gems/chef-12.4.0-universal-mingw32 for inspection.
Results logged to W:/home/.chefdk/gem/ruby/2.1.0/extensions/x86-mingw32/2.1.0/chef-12.4.0-universal-mingw32/gem_make.out
An error occurred while installing chef (12.4.0), and Bundler cannot continue.
Make sure that `gem install chef -v '12.4.0'` succeeds before bundling.
```

**Comment by tknerr** [comment]
Might be related to #3125, which was fixed with chef 12.3.0 via #3126 though

No clue why this one suddenly occurs... (might be a dependendent gem that changed, or even windows updates?!?)

**Comment by jaym** [comment]
you're not locking:
```
   80 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/2.1/ffi_c.so
   81 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/platform.rb
   82 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/types.rb
   83 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/library.rb
   84 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/errno.rb
   85 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/pointer.rb
   86 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/memorypointer.rb
   87 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/struct_layout_builder.rb
   88 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/struct.rb
   89 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/union.rb
   90 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/managedstruct.rb
   91 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/callback.rb
   92 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/io.rb
   93 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/autopointer.rb
   94 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/variadic.rb
   95 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/enum.rb
   96 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi/ffi.rb
   97 W:/home/.chefdk/gem/ruby/2.1.0/gems/ffi-1.9.9-x86-mingw32/lib/ffi.rb
   98 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/windows/structs.rb
   99 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/windows/functions.rb
  100 W:/home/.chefdk/gem/ruby/2.1.0/gems/win32-eventlog-0.6.3/lib/win32/eventlog.rb
```

**Comment by tknerr** [comment]
@jaym just saw that too. In fact I added `gem 'ffi', '1.9.8'` to my Gemfile but it didn't pick it up. Can reproduce it with 1.9.8 being locked in the Gemfile and 1.9.9 being present in `gem list`

**Comment by tknerr** [comment]
After `gem uninstall ffi --version 1.9.9` the `bundle install` passes now

**Comment by tknerr** [comment]
So it's definitely a variation of #3606, and I can totally reproduce it:

* `gem uninstall chef --version 12.4.0`
* `gem install ffi --version 1.9.9`

Then, using this Gemfile (note ffi being locked to 1.9.8) I get the segfault above on `bundle install`:
```
source 'https://rubygems.org'
gem 'ffi', '1.9.8'
gem 'chef', '12.4.0'
```

**Comment by tknerr** [comment]
So it seems that even though I lock to 1.9.8 in the Gemfile chef uses the latest available when it tries to register the event log source in the Rakefile here:
https://github.com/chef/chef/blob/master/ext/win32-eventlog/Rakefile#L45

Not sure if the fix in #3606 helps if a user has ffi 1.9.9 installed already, let me check...

**Comment by tknerr** [comment]
Tried `bundle install` with chef 12.4.0.dev from master (which includes #3606) with this Gemfile:
```
source 'https://rubygems.org'
gem 'chef', '12.5.0.current.0', git: 'https://github.com/chef/chef', ref: 'master'
```

=> segfault still occurs, and it still uses ffi 1.9.9 if present :-(

**Comment by tknerr** [comment]
@jaym is there any way to ensure that the Rakefile which registers the event source uses the desired ffi <1.9.9 version? 

Maybe need to add it as a development dependency as well in the gemspec?

**Comment by jaym** [comment]
perhaps change `system 'rake blah'` to `system 'bundle exec rake blah'` in the Rakefile?

**Comment by tknerr** [comment]
@jaym which place you mean? Not sure where [this Rakefile](https://github.com/chef/chef/blob/master/ext/win32-eventlog/Rakefile#L45) is being called from...

btw: `add_development_dependency` did not help...

**Comment by tknerr** [comment]
@jaym it's probably not a problem in omnibus chef as long as you ensure there is no ffi 1.9.9 included

Guess this is likely to happen only if you install chef as a gem... 

I thought of adding Gemfile right beside the Rakefile and doing a `require 'bundler/setup'`, but not sure if this is a good idea...

**Comment by jaym** [comment]
```diff
diff --git a/Rakefile b/Rakefile
index 628e0e1..88b18ca 100644
--- a/Rakefile
+++ b/Rakefile
@@ -143,13 +143,13 @@ end
 
 task :build_eventlog do
   Dir.chdir 'ext/win32-eventlog/' do
-    system 'rake build'
+    system 'bundle exec rake build'
   end
 end
 
 task :register_eventlog do
   Dir.chdir 'ext/win32-eventlog/' do
-    system 'rake register'
+    system 'bundle exec rake register'
   end
 end
```

**Comment by tknerr** [comment]
@jaym this gets it working for my specific case, but it probably does not work as expected

Why? Because it uses [my Gemfile](https://github.com/chef/chef/issues/3609#issuecomment-116365501) rather than the included [chef/chef/Rakefile.rb](https://github.com/chef/chef/blob/master/Gemfile).

So I can now fixate the ffi dependency from the outside, but the ffi version constrained from the chef gemspec is still being ignored.

Running out of ideas.... *sigh*

**Comment by lamont-granquist** [comment]
Ruby 2.1.x on windows is also known to be busted, that's why omnibus-chef windows is still pinned at 2.0.0

**Comment by tknerr** [comment]
@lamont-granquist yep, I'm using exactly the Chef-DK 2.1 Ruby... ;-)

Apart from [that](https://bugs.ruby-lang.org/issues/11033) I was not seeing obvious problems yet. If you can point me to more known Ruby 2.1 / windows quirks I'd be interested in that too

**Comment by lamont-granquist** [comment]
ffi 1.9.10 is out now, fixed the problem?

**Comment by lamont-granquist** [comment]
@jaym might know more offhand about the general issues with ruby 2.1 and windows and chef-client.

**Comment by jaym** [comment]
This was the whole bundler going crazy when you told it to use multiple threads. I think @tknerr was the one who reported it

**Comment by tknerr** [comment]
@jaym @lamont-granquist yeah that's the one I'm aware of:
https://bugs.ruby-lang.org/issues/11033

Also, the issue is fixed with ffi 1.9.10

So the general problem remains (can not lock the ffi version during `bundle install` of chef), but the concrete issue (segfault) is fixed now that ffi 1.9.10 is out

Should we close this issue then?

**Comment by lamont-granquist** [comment]
Yeah, I think we're at least aware of the bundler threaded yakshave and this probably isn't the ticket to work that problem.  The ffi segfault is fixed by 1.9.10 so this should be closed.  If there needs to be another ticket on the bundler stuff for us that would most likely be an issue that needs to get cut in the omnibus-chef repo for when the core bundler issue gets resolved.
